src/Riok.Mapperly/Descriptors/Mappings/MethodMapping.cs

```diff
@@ -33,6 +33,7 @@ public abstract class MethodMapping : NewInstanceMapping
     private readonly IMethodSymbol? _partialMethodDefinition;
 
     private string? _methodName;
+    private bool? _methodIsStatic;
```

Comment:
I don't think this approach does fully work. Scenario where I think it wont work (probably add a unit test for this):
* Non-static mapper class
* Source object tree `A>B>C>int`
* Target object property tree `D>E>F>int`.
* User defined mapping method from A to D
* User implemented mapping method from C to F

Mapperly generates the B to E method as an instance method. From this method it then calls the user implemented static method to convert from C to F. I think whether the mapper is a static/pseudo-static mapper needs to be extracted into the descriptor and passed to the `SourceEmitterContext` in `Riok.Mapperly.Emit.SourceEmitter.Build`.

Reply:
I changed it to SourceEmitterContext version, and I added some integration tests:
<img width="779" alt="image" src="https://github.com/riok/mapperly/assets/3594540/2b8964ef-a736-46a0-8e29-efca6d1e8725">
I hope it tests what you mentiond

Reply:
IMO this should rather be a unit test then an integration test. It's behaviour doesn't really need to be tested on different .net versions (that's mainly what integration tests are for).

Reply:
I use it because it compiles generated code -> discover instance method calls from static methods. But I can move it to unit test also

Reply:
Do you want me to drop this part?
Unit tests were updated

Reply:
Now this is unit tested quite well and Mapperly runs the same code for all Roslyn versions, therefore I don't see a need to keep this additional integration test: I'd drop it.

Reply:
I dropped this integration test